dart/lib/src/hint.dart

```diff
@@ -0,0 +1,27 @@
+class Hint {
+  final Map<String, Object> _internalStorage = {};
+
+  Hint();
+
+  Hint.fromMap(Map<String, Object> map) {
+    _internalStorage.addAll(map);
+  }
```

Comment:
Do you wanna add new items only or creating a new `Hint`? if a new `Hint`, it'd make more sense to be a `factory`